apis/system/tests/common.py

```diff
@@ -0,0 +1,39 @@
+import subprocess
+
+
+def to_R(x):
+    if isinstance(x, bool):
+        return str(x).upper()
+    elif isinstance(x, str):
+        return f'"{x}"'
+    else:
+        return str(x)
+
+
+def create_R_list(xs):
+    return ",".join([to_R(x) for x in xs])
+
+
+class TestWritePythonReadR:
+    @classmethod
+    def setup_class(cls):
+        import tempfile
+
+        cls.td = tempfile.TemporaryDirectory()
```

Comment:
please have `cls.td` -> `cls.tempdir` throughout. much easier to read/understand for everyone after you.